auth0/src/main/java/com/auth0/android/provider/WebAuthProvider.java

```diff
@@ -57,17 +60,107 @@
 
-    private static final String KEY_AUDIENCE = "audience";
-    private static final String KEY_SCOPE = "scope";
-    private static final String KEY_CONNECTION_SCOPE = "connection_scope";
-    private static final String SCOPE_TYPE_OPENID = "openid";
-    private static final String RESPONSE_TYPE_TOKEN = "token";
+    private static ResumableManager managerInstance;
 
-    private static OAuthManager managerInstance;
+    public static class LogoutBuilder {
 
+        private static final String KEY_FEDERATED = "federated";
+        private static final String KEY_RETURN_TO_URL = "returnTo";
+
+        private final Auth0 account;
+        private final Map<String, String> values;
+        private String scheme;
+        private CustomTabsOptions ctOptions;
+
+        LogoutBuilder(Auth0 account) {
+            this.account = account;
+
+            //Default values
+            this.scheme = "https";
+            this.values = new HashMap<>();
+        }
+
+        /**
+         * When using a Custom Tabs compatible Browser, apply this customization options.
+         *
+         * @param options the Custom Tabs customization options
+         * @return the current builder instance
+         */
+        public LogoutBuilder withCustomTabsOptions(@NonNull CustomTabsOptions options) {
+            this.ctOptions = options;
+            return this;
+        }
+
+        /**
+         * Specify a custom Scheme to use on the Return To Uri. Default scheme is 'https'.
+         *
+         * @param scheme to use in the Callback Uri.
+         * @return the current builder instance
+         */
+        public LogoutBuilder withScheme(@NonNull String scheme) {
+            String lowerCase = scheme.toLowerCase();
+            if (!scheme.equals(lowerCase)) {
+                Log.w(TAG, "Please provide the scheme in lowercase and make sure it's the same configured in the intent filter. Android expects the scheme in lowercase");
+            }
+            this.scheme = scheme;
+            return this;
+        }
+
+        /**
+         * Whether to also clear the session from the identity provider when possible or not.
+         *
+         * @param federatedLogout true to clear the session on the identity provider's site.
+         * @return the current builder instance
+         */
+        public LogoutBuilder useFederatedLogout(@NonNull boolean federatedLogout) {
+            if (federatedLogout) {
+                this.values.put(KEY_FEDERATED, "1");
+            } else {
+                this.values.remove(KEY_FEDERATED);
+            }
+            return this;
+        }
+
+        /**
+         * Request the user session to be cleared. When successful, the callback will get invoked
+         *
+         * @param context  to run the log out
+         * @param callback to invoke when log out is successful
+         */
+        public void start(Context context, BaseCallback<Void, Auth0Exception> callback) {
```

Comment:
Feedback on the method name?
e.g. start or launch? The auth one uses "start"

Reply:
I think `start` is fine, and it's consistent.